assets/hdfs_append/index.js

```diff
@@ -21,6 +21,68 @@ function getClient(context, config, type) {
     return context.foundation.getConnection(clientConfig);
 }
 
+// Records the name of the offending file with a detected corrupt block in `appendErrors`
+function _recordFileError(name, errorLog) {
+    let newFilename = '';
+    // Get the original file name from the error message. If there have been no previous errors
+    // for the given file, this will result in a bogus filename that will trigger the second
+    // block of the following checks. Otherwise, the third block will be used to get a new name
+    const fileBase = name
+        .split('.')
+        .reverse()
+        .splice(1)
+        .reverse()
+        .join('.');
+    if (!errorLog.retry) {
+        // In this case, there has not been any hdfs append errors for a worker
+
+        newFilename = `${name}.0`;
+        errorLog[name] = newFilename;
+        // Ensures this block will not be executed again after the first error for a file
+        errorLog.retry = true;
+    } else if (!errorLog[fileBase]) {
+        // In this case, there has been at least one hdfs append error for a worker, but the
+        // worker is now writing to a new file
+
+        // Clean out the appendErrors object to avoid an accumulation of filenames
+        Object.keys(errorLog).forEach((key) => {
+            delete errorLog[key];
+        });
+
+        newFilename = `${name}.0`;
+        errorLog[name] = newFilename;
+        // Add the `retry` key back in
+        errorLog.retry = true;
+    } else {
+        // In this case, there has been at least one hdfs append error for the given file
+
+        // Get the last attempted file and increment the number
+        const incNum = errorLog[fileBase].split('.').reverse()[0] * 1 + 1;
```

Comment:
I use that to make sure the result is an integer instead of a string. Would it be a better practice to do something like `const incNum = parseInt(errorLog[fileBase].split('.').slice(-1)[0]) + 1` in this case?

Reply:
You can do `+errorLog[fileBase].split('.').slice(-1)`. Note no `[0]` required when using slice.

Of course ... this is going to blow up if that extension is not a number for some reason.

Reply:
Cleaned this up with suggested fix. This extension will always be an integer since this portion of the code will only be executed if there has been a previous error, and the value it is referencing will always end with the error counter added to it at the last append error.